### DISCUSSION TITLE
GraphQL asset interface does not expose mp4/video dimensions

### DISCUSSION THREAD

**Discussion by riccardolardi** [craftcms/cms#9038] [Ideas]
### Description

Is there any chance to implement this in near future and in the meantime is there any workaround to retrieve mp4/video dimensions in any other way? 
I'm working on a masonry frontend which consists of images and videos where every asset can have different dimensions. Currently I can't seem to get dimensions data on the mp4 video assets through GraphQL (width and height fields return null).

**Comment by andris-sevcenko** [comment]
That's not a GraphQL limitation, though, as Craft doesn't have the capability to figure out the dimensions for a video file, yet.

If the values are in the `assets` table, they will get returned, so a possible solution is to populate those fields manually in the database.

**Reply by nodoxi** [reply]
@andris-sevcenko 
it's GraphQL limitation.
I edited `assets` table and added width and height to mp4/video asset and still getting `null` in GraphQL.
![image](https://user-images.githubusercontent.com/2776595/140279727-82b50003-2d19-4913-9f28-c3700c3a950f.png)

GraphQL
![image](https://user-images.githubusercontent.com/2776595/140279839-edad48d1-0eee-4b98-b704-47a192ff04be.png)

**Reply by andris-sevcenko** [reply]
Right, you would get the same result in a Twig template, so it's a Craft limitation, not GraphQL.

Currently, Craft [doesn't return dimensions for anything that is not an image](https://github.com/craftcms/cms/blob/96f1a446a1e513e8b545cf822007395227d99c77/src/elements/Asset.php#L2155) (contrary to what I wrote before, apologies for that mistake).

There are some caveats to navigate - we'll talk this over internally and see if we can safely just return whatever is stored in the database.

**Reply by andris-sevcenko** [reply]
Just added this for the next Craft 3.7 release (738906772dc).

**Reply by brandonkelly** [reply]
Craft 3.7.21 is out now with that change.
